src/editor.c

```diff
@@ -2458,7 +2457,7 @@ void editor_insert_text_block(GeanyEditor *editor, const gchar *text, gint inser
 	gint line_start = sci_get_line_from_position(sci, insert_pos);
 	GString *buf;
 	const gchar *eol = editor_get_eol_char(editor);
-	gint idx;
+	gint idx = 0;
```

Comment:
Looking at it more closely, the issue is more serious, and your fix seems insufficient. I'll look into it.